python/lsst/ts/dream/common/index_generator.py

```diff
@@ -0,0 +1,70 @@
+# This file is part of ts_dream_common.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["index_generator"]
+
+import typing
+
+# Maximum allowed SAL index (inclusive)
+MAX_INDEX = (1 << 31) - 1
+
+
+def index_generator(
```

Comment:
isn't this part of ts_utils now?

Reply:
Yes it is now though it wasn't at the time that I put the PR in review. I have updated the code to make use of the ts_utils `index_generator`.